### ISSUE TITLE
Not allowed to use "flex" as a prop (shadow node)

### ISSUE THREAD

**Issue by RWOverdijk** [facebook/react-native#33411]
### Description

https://github.com/facebook/react-native/issues/23522

This issue was closed, but this seems to still be an issue on android (iOS works fine).

To be very clear (because of the comment on that issue): I know that flex is a style prop. I know it's not a View prop.

But apparently it _is_ a View prop on android.

The error:

> Error while updating property 'flex' in shadow node of type: RCTView. java.lang.Boolean cannot be cast to java.lang.Double

My problem with this is that this prop isn't documented and is causing crashes on android that I found through logcat. I think this prop should either be removed or documented.

### Version

0.64.3

### Output of `npx react-native info`

System:
    OS: macOS 12.2.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 86.98 MB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.13.1 - ~/.nvm/versions/node/v16.13.1/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v16.13.1/bin/yarn
    npm: 8.1.2 - ~/.nvm/versions/node/v16.13.1/bin/npm
    Watchman: 2021.12.20.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: Not Found
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    Android SDK:
      API Levels: 23, 28, 29, 32
      Build Tools: 28.0.3, 29.0.2, 30.0.2, 32.0.0
      System Images: android-29 | Intel x86 Atom_64, android-29 | Google Play ARM 64 v8a, android-30 | Google Play ARM 64 v8a, android-32 | Google APIs ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2020.3 AI-203.7717.56.2031.7935034
    Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
  Languages:
    Java: 17.0.2 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1
    react-native: 0.64.3 => 0.64.3
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

It's easiest to look at the snack I made here: https://snack.expo.dev/@rwoverdijk/bossy-sandwich

The tl;dr; code is:

```tsx
const Container = styled.View`
  ${p => p.flex && 'flex: 1;'}
`;

export default () => (
  <Container flex /> // <-- Crashes android
);
```

### Snack, code example, screenshot, or link to a repository

This snack:

https://snack.expo.dev/@rwoverdijk/bossy-sandwich

Works fine in the browser and crashes expo-go on android (I thougt this might be the easiest way to test this).

Removing "flex" from "<Container flex>" stops it from crashing.

**Comment by haveneersrobin** [comment]
It could also be this piece code that is causing the issue `p.flex && 'flex: 1;'`
See the error: `RCTView. java.lang.Boolean cannot be cast to java.lang.Double`
Have you tried: `${p => p.flex ? 'flex: 1;' : undefined}`

Isn't it also possible that Styled Components is doing a faulty conversion? What happens if you use `<Container flexSomething />` ?

**Comment by RWOverdijk** [comment]
Good question! `flexSomething` works fine, it's only the prop flex and only on android

**Comment by haveneersrobin** [comment]
Seems like an issue with styled components then :) I would suggest opening an issue there, or using a non-'_reserved_' word

**Comment by RWOverdijk** [comment]
I disagree. If it's reserved it should be documented. I am already using a non-"reserved" prop, but that doesn't mean this undocumented magic is suddenly not an issue 😄

**Comment by haveneersrobin** [comment]
But, this is an issue with styled components then, not with `react-native`. 

Your
``` ts
const Container = styled.View`
  ${p => p.flex && 'flex: 1;'}
`;
```
is a styled-component, not a regular component. Whatever magic they do and props they add/transform/whatever: that is where this goes wrong.

Also, if you use TypeScript, you will get an error (at least with the regular RN View)

![image](https://user-images.githubusercontent.com/7559898/158822761-2096b80d-9d0b-4668-98fb-87958208e56e.png)

**Comment by RWOverdijk** [comment]
I know what it is. I also understand what you're saying. I just disagree with you. Feel free to close this if you think a magic property is not an issue. Because:

> Also, if you use TypeScript, you will get an error (at least with the regular RN View)

Yes, you're right. That just underlines what I'm saying. It doesn't exist, so it shouldn't panic. It's undocumented behaviour. That should be remedied.

Regarding the repeated point about "styled-components": I did that. I tried that. But I found this before I made this issue:

https://github.com/styled-components/styled-components/issues/3190#issuecomment-654857251

> The start prop there seems to actually be picked up by React Native on the builtin component you're rendering. Since props aren't filtered in styled-components for React Native you must make sure that your props never overlap with props that the component accepts.

They're probably not going to rewrite the core logic because I want to use a prop named `flex`. So the tl;dr;

- The prop shouldn't cause crashes based on undocumented behaviour **<-- specifically what this issue is about.**
- Styled components shouldn't do whatever it is doing passing these props to the underlying components.
- Alternatively I should make my own component but then why use styled-components to begin with.
